ocean_lib/ocean/ocean_assets.py

```diff
@@ -319,6 +359,17 @@ def create(
 
         return asset
 
+    # I need an update function for trusted_algorithms
+    # that I will call in test_compute_flow.py
+    def update_trusted_algorithms(self, trusted_algorithms=None):
+        services = self._process_service_descriptors(
+            self.service_descriptors,
+            self.metadata_copy,
+            self.provider_uri,
+            self.publisher_wallet,
+            trusted_algorithms=trusted_algorithms,
+        )
```

Comment:
I think this is a good start, but it is still incomplete. See how services are attached in the create function and try to mimic that. Keep all previous services of the asset in place, but set/change the compute service's trusted algorithms. I think that is the general idea, but more guidance from @ssallam is always welcome! :D